domain/user/repository/postgre/user.go

```
@@ -77,3 +78,17 @@ func (p *userRepoImpl) VerifyUserByEmail(ctx context.Context, email string) (boo
 	}
 	return true, nil
 }
+
+func (p *userRepoImpl) GetByTokens(ctx context.Context, params *entity.GetByTokensRequest) (*entity.User, error) {
+	var user *entity.User
+	err := p.db.
+		Where("id = ?", params.UserId).
+		Where("(refresh_token = ? OR access_token = ?)", params.Token, params.Token).
+		First(&user).Error
+
+	if errors.Is(err, gorm.ErrRecordNotFound) {
```

I think it's still better to return this as error, and handle it in service package mas, and while on that perhaps you could also delete the brackets inside refresh_token = ? or access_token = ? query